packages/@orbit/jsonapi/src/jsonapi-serializer.ts

```diff
@@ -309,6 +326,111 @@ export class JSONAPISerializer
     return result;
   }
 
+  deserializeOperationsDocument(
+    document: ResourceOperationsDocument
+  ): RecordOperation[] {
+    return this.deserializeOperations(document.operations);
+  }
+
+  deserializeOperations(operations: ResourceOperation[]): RecordOperation[] {
+    return operations.map(operation => this.deserializeOperation(operation));
+  }
+
+  deserializeOperation(operation: ResourceOperation): RecordOperation {
+    if (isAddOperation(operation)) {
+      return this.deserializeAddOperation(operation);
+    } else if (isUpdateOperation(operation)) {
+      return this.deserializeUpdateOperation(operation);
+    } else if (isRemoveOperation(operation)) {
+      return this.deserializeRemoveOperation(operation);
+    } else {
+      throw new Error(
+        `JSONAPISerializer: "get" operation recieved but only "add", "update" and "remove" operations are supported as input at this time.`
+      );
+    }
+  }
+
+  deserializeAddOperation(
+    operation: AddResourceOperation | AddToRelatedResourcesOperation
+  ): AddRecordOperation | AddToRelatedRecordsOperation {
+    if (isRelatedResourceOperation(operation)) {
+      return {
+        op: 'addToRelatedRecords',
+        relationship: operation.ref.relationship,
+        record: this.deserializeResourceIdentity(operation.ref),
+        relatedRecord: this.deserializeResourceIdentity(
+          operation.data as RecordIdentity
+        )
+      };
+    } else {
+      return {
+        op: 'addRecord',
+        record: this.deserializeResource(operation.data)
+      };
+    }
+  }
+
+  deserializeUpdateOperation(
+    operation:
+      | UpdateResourceOperation
+      | ReplaceRelatedResourceOperation
+      | ReplaceRelatedResourcesOperation
```

Comment:
Is the leading `|` form a common TS convention that I've never noticed? I assume the first one has no impact. Is there a reason to go with this instead of:

```
operation:
  UpdateResourceOperation |
  ReplaceRelatedResourceOperation |
  ReplaceRelatedResourcesOperation
```

Reply:
hmm I agree it's a bit wired. I haven't noticed it, but this is the result of running prettier...

Reply:
ok, I guess we'll live with it then 🤷‍♂